addons/mail/models/res_partner.py

```diff
@@ -20,6 +22,15 @@ class Partner(models.Model):
     vat = fields.Char(tracking=5)
     # channels
     channel_ids = fields.Many2many('mail.channel', 'mail_channel_member', 'partner_id', 'channel_id', string='Channels', copy=False)
+    # add tracked field
+    contact_address_inline = fields.Char(compute='_compute_contact_address_inline', string='Inlined Complete Address', tracking=True)
+
+    @api.depends('contact_address')
+    def _compute_contact_address_inline(self):
+        """Compute an inline-friendly address, not guaranteed to be a valid postal format."""
+        for partner in self:
+            # replace any successive \n by a single comma
+            partner.contact_address_inline = re.sub(r'\n( *\n)*', ', ', partner.contact_address)
```

Comment:
I know it's a simple regex but it would be nice to add a test.
(Looking at validation comments, it looks planned).

Reply:
Added a test for the regex. The same test also tests the additional behaviour lna wanted, i.e. that we avoid logging address changes in company employees if their address is determined by the company.

From what I've seen the intended behavior is that the address fields are 'related' if both the parent and child are 'contact' types. I don't think it's actually enforced by the ORM though, so maybe we should just compare with the parent to be sure? Let me know